### PR TITLE
emit @private/@protected where needed

### PR DESCRIPTION
**Pull request by evmar**
These not only matter for downstream JS code, but also appear to
matter for some optimizations.

### CODE REVIEW

**Comment by evmar** [comment]
Chances are low that this will work in google3...

File: test_files/protected/protected.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/** This test checks that we emit @private/@protected where necessary. */
```
Comment (mprobst) [inline test_files/protected/protected.ts]: nit: should you escape the `\@private`? I never know how escaping even works in JSDoc though...

**Comment by evmar** [comment]
I escaped the @private just in case, and also had to adjust the test runner to not fail on unused private fields.

**Comment by evmar** [comment]
PTAL, I had to add a new @suppress.

**Comment by evmar** [comment]
Interestingly, the failing e2e test tipped me off to this -- adding more flag changes to our e2e test is usually wrong, because we need our emit to be accepted by whatever flags users have set.

**Pull request merged**